package/service/submission_service.go

```diff
@@ -13,17 +15,186 @@ type SubmissionService interface {
 	MarkSubmissionFailed(tx *gorm.DB, submissionId int64, errorMsg string) error
 	MarkSubmissionComplete(tx *gorm.DB, submissionId int64) error
 	MarkSubmissionProcessing(tx *gorm.DB, submissionId int64) error
+	CreateSubmission(tx *gorm.DB, taskId int64, userId int64, languageId int64, order int64) (int64, error)
 	CreateSubmissionResult(tx *gorm.DB, submissionId int64, responseMessage schemas.ResponseMessage) (int64, error)
+	GetAll(tx *gorm.DB, user schemas.User, queryParams map[string][]string) ([]schemas.Submission, error)
+	GetById(tx *gorm.DB, submissionId int64, user schemas.User) (schemas.Submission, error)
+	GetAllForUser(tx *gorm.DB, userId int64, user schemas.User, queryParams map[string][]string) ([]schemas.Submission, error)
+	GetAllForGroup(tx *gorm.DB, groupId int64, user schemas.User, queryParams map[string][]string) ([]schemas.Submission, error)
+	GetAllForTask(tx *gorm.DB, taskId int64, user schemas.User, queryParams map[string][]string) ([]schemas.Submission, error)
 }
 
 type SubmissionServiceImpl struct {
 	submissionRepository       repository.SubmissionRepository
 	submissionResultRepository repository.SubmissionResultRepository
 	inputOutputRepository      repository.InputOutputRepository
 	testResultRepository       repository.TestResultRepository
+	userService                UserService
+	taskService                TaskService
+	languageService            LanguageService
 	logger                     *zap.SugaredLogger
 }
 
+var ErrPermissionDenied = errors.New("User is not allowed to view this submission")
+
+func (us *SubmissionServiceImpl) GetAll(tx *gorm.DB, user schemas.User, queryParams map[string][]string) ([]schemas.Submission, error) {
```

Comment:
Implementation is ok, but I think that we will not use this function as intended. What I think is that we will split this into 3 different service methods. Something like:
`GetMy` - retrieves submissions only submitted by this user. For example stuent can see all his submissions
`GetForTask` -  retreives submissions only for specific task, used by the teacher
`GetForTeacher` -  retrieves submissions for all tasks created by this teacher
`GetAll` - retrieves all submissions, for admins
This is more or less what you did, but split into smaller version of functions. 

You don't need to implement this now though. I will do this after my planned API refactor